lib/composite_cache_store.rb

```diff
@@ -4,146 +4,164 @@
 require_relative "composite_cache_store/version"
 
 class CompositeCacheStore
```

Comment:
Okay, let me just ask this, and if the answer's "no" I'll offer thoughts on other things :)

Would it work for this to be a subclass of `ActiveSupport::Cache::Store`, and see how close we could get by overriding `initialize` and the three "must implement" methods?

In particular, to see if we don't need to override `fetch` etc.? I'm staring at the logic of those methods you implemented and at the abstract class's implementations, and I'm not sure why just overriding the low-level methods couldn't work.

Reply:
I may experiment with this again and will let you know how that effort goes.

Reply:
I've been tinkering on this a bit but am going to punt on a refactor that moves away from composition to inheritance for now. I'm open to exploring this strategy in another PR though.